common/types/shared.go

```diff
@@ -465,6 +677,42 @@ func (v *ChildWorkflowExecutionCanceledEventAttributes) GetInitiatedEventID() (o
 	return
 }
 
+// Size returns the approximate memory used in bytes
+func (v *ChildWorkflowExecutionCanceledEventAttributes) Size() uint64 {
+	if v == nil {
+		return 0
+	}
+
+	size := uint64(40)
+
+	if v.Details != nil {
+		size += uint64(len(v.Details))
+	}
+
+	if v.Domain != "" {
+		size += uint64(16 + len(v.Domain))
+	}
+
+	if v.WorkflowExecution != nil {
+		size += 24
+		if v.WorkflowExecution.WorkflowID != "" {
+			size += uint64(16 + len(v.WorkflowExecution.WorkflowID))
+		}
+		if v.WorkflowExecution.RunID != "" {
+			size += uint64(16 + len(v.WorkflowExecution.RunID))
+		}
```

Comment:
you are doing this calculation on `WorkflowExecution` type field in multiple places. Just expose `Size()` on that type and call it.

Reply:
let's make sure we avoid redundant calculations like this for other typed fields